src/lists_ext.erl

```diff
@@ -0,0 +1,44 @@
+-module(lists_ext).
+
+-export([ zip_uneven_lists/2,
+          zip_uneven_lists/3,
+          zip_uneven_lists_stable/2,
+          zip_uneven_lists_stable/3
+        ]).
+
+-spec zip_uneven_lists(list(), list()) -> list().
+zip_uneven_lists(L1, L2) -> zip_uneven_lists(L1, L2, undefined).
+
+-spec zip_uneven_lists(list(), list(), any()) -> list().
+zip_uneven_lists(L1, L2, _Default) when length(L1) =:= length(L2) -> lists:zip(L1, L2);
+zip_uneven_lists(L1, L2, Default) when length(L1) > length(L2)    -> zip_with_default(L1, L2, Default);
+zip_uneven_lists(L1, L2, Default) when length(L1) < length(L2)    -> zip_with_default(L2, L1, Default).
```

Comment:
IMHO changing order of lists is a little bit counter-intuitive and changes order in result tuples:
```
lists_ext:zip_uneven_lists(["A1"], ["B1", "B2"]).
[{"B1","A1"},{"B2",undefined}]
```
I would be better if first element of resulting tuples was always from first list passed to the function, thus default and only desired behavior is the one from `zip_with_default_stable/3`